common/src/main/java/com/wynntils/utils/StringUtils.java

```diff
@@ -65,4 +65,13 @@ public static String[] wrapTextBySize(String s, int maxPixels) {
 
         return result.toString().split("\n");
     }
+
+    /**
+     * Creates a new pattern, but replaces all occurrences of '§' in the regex with an expression for detecting any color code
+     * @param regex - the expression to be modified and compiled
+     * @return a Pattern with the modified regex
+     */
+    public static Pattern compileCCRegex(String regex) {
```

Comment:
I don't really know where to put this method. Also, not sure about the name either.

Reply:
That was a clever idea to get simple but proper matching for color codes. Did you come up with it or was it in legacy? Also, it's a good thing to move it to StringUtils, I do think it belongs here.

Maybe you should strip it down to the bare str.replace() part, and leave the Pattern.compile to the caller. Then you can name it like `replaceColorCode`. I think that would do fine.

Reply:
The idea was from legacy code.
The main reason was for ccregex creation to be shorter, so spotless formatting won't make it a 2-liner. I'm not a big fan of the replace-only idea, as this method doesn't really make sense outside of regex creation. I'll think about it.